src/main/java/io/vertx/core/file/impl/FileResolverImpl.java

```diff
@@ -295,17 +296,28 @@ private File unpackFromJarURL(URL url, String fileName, ClassLoader cl) {
         zip = new ZipFile(file);
       }
 
+      String inJarPath = path.substring(idx1 + 6);
+      StringBuilder prefixBuilder = new StringBuilder();
+      int first = 0;
+      int second;
+      int len = JAR_URL_SEP.length();
+      while ((second = inJarPath.indexOf(JAR_URL_SEP, first)) >= 0) {
+        prefixBuilder.append(inJarPath, first, second).append("/");
+        first = second + len;
+      }
+      String prefix = prefixBuilder.toString();
       Enumeration<? extends ZipEntry> entries = zip.entries();
       while (entries.hasMoreElements()) {
         ZipEntry entry = entries.nextElement();
         String name = entry.getName();
-        if (name.startsWith(fileName)) {
+        if (name.startsWith(prefix.isEmpty() ? fileName : prefix + fileName)) {
```

Comment:
the expression `prefix.isEmpty() ? fileName : prefix + fileName)` can be moved out of the while loop which avoids to compute prefix + fileName at each iteration

Reply:
Done